homeassistant/components/pilight.py

```diff
@@ -0,0 +1,116 @@
+"""
+Component to create an interface to a Pilight daemon (https://pilight.org/).
+
+Pilight can be used to send and receive signals from a radio frequency
+module (RF receiver).
+
+RF commands received by the daemon are put on the HA event bus.
+RF commands can also be send with a pilight.send service call.
+"""
+# pylint: disable=import-error
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.helpers import validate_config
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import ensure_list
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+REQUIREMENTS = ['pilight==0.0.2']
+
+DOMAIN = "pilight"
+EVENT = 'pilight_received'
+SERVICE_NAME = 'send'
+
+ATTR_PROTOCOL = 'protocol'
+
+# The pilight code schema depends on the protocol
+# Thus only require to have the protocol information
+RF_CODE_SCHEMA = vol.Schema({vol.Required(ATTR_PROTOCOL): cv.string},
+                            extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setup pilight component."""
+    from pilight import pilight
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_HOST, CONF_PORT]},
+                           _LOGGER):
+        return False
+
+    try:
+        pilight_client = pilight.Client(host=config[DOMAIN][CONF_HOST],
+                                        port=config[DOMAIN][CONF_PORT])
+    except (socket.error, socket.timeout) as err:
+        _LOGGER.error(
+            "Unable to connect to %s on port %s: %s",
+            config[CONF_HOST], config[CONF_PORT], err)
+        return False
+
+    # Start / stop pilight-daemon connection with HA start/stop
+    def start_pilight_client(_):
+        """Called once when home assistant starts."""
+        pilight_client.start()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_pilight_client)
+
+    def stop_pilight_client(_):
+        """Called once when home assistant stops."""
+        pilight_client.stop()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_pilight_client)
+
+    def send_code(call):
+        """Send RF code to the pilight-daemon."""
+        message_data = call.data
+
+        if "protocol" not in message_data:
+            _LOGGER.error(
+                'Pilight data to send does not contain a protocol info: %s.'
+                ' Check the pilight-send documentation!',
+                str(call.data))
+            return
+
+        # Patch data because of bug:
+        # https://github.com/pilight/pilight/issues/296
+        # Protocol has to be in a list otherwise segfault in pilight-daemon
+        message_data["protocol"] = ensure_list(message_data["protocol"])
```

Comment:
I think to implement all available protocols is not a solution here. There are too many and this would require active maintenance for the HA component.